src/App.tsx

```diff
@@ -7,15 +8,24 @@ import { createBrowserRouter, RouterProvider } from 'react-router-dom'
 import { ROUTES } from './routes'
 import Disclaimer from './shared/components/Disclaimer'
 
+const isProd = process.env.NODE_ENV === 'production'
+const siteId = process.env.REACT_APP_PIWIK_SITE_ID
+
 const queryClient = new QueryClient({
   defaultOptions: {
     queries: {
-      retry: process.env.NODE_ENV === 'production' ? 3 : false,
-      refetchOnWindowFocus: process.env.NODE_ENV === 'production',
+      retry: isProd ? 3 : false,
+      refetchOnWindowFocus: isProd,
     },
   },
 })
 
+const piwikInstance = createInstance({
+  urlBase: process.env.REACT_APP_SELF_ROOT,
+  siteId: siteId ?? '0',
+  disabled: !isProd || !siteId,
+})
+
 const router = createBrowserRouter(ROUTES)
 
 function App() {
```

Comment:
In een ander project (Mijn Amsterdam) hebben we een `usePageChange` hook gemaakt. Daarin zit een `useEffect` die als dependencies de `location.pathname` en `location.search` heeft. In de het effect wordt dan de `trackPageView` functie van de `PiwikInstance` aangeroepen .
De hook aanroepen zou je in het `App` component kunnen doen. Gevolg is dat pagina's geen `AnalticsProvider` meer nodig hebben.

Reply:
Ik heb je suggestie met aanpassingen doorgevoerd. Ipv useEffect gebruikt de app een useCallback om expliciet per pagina te visit te tracken.